src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp

```diff
@@ -339,6 +368,127 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
   return true;
 }
 
+bool EinsplineSetBuilder::ReadGvectors_ESHDF()
+{
+  bool root = myComm->rank() == 0;
+  //this is always ugly
+  MeshSize    = 0;
+  int hasPsig = 1;
+  if (root)
+  {
+    H5File.readEntry(MeshSize, "/electrons/psi_r_mesh");
+    H5File.readEntry(MeshSize, "/electrons/mesh");
+  }
+  myComm->bcast(MeshSize);
+  hasPsig = (MeshSize[0] == 0);
+  if (hasPsig)
+  {
+    int nallowed  = 257;
```

Comment:
Can you explain + justify this please?

Reply:
I think this was initially introduced to block bad fft grids in certain libraries (IIRC essl).

Reply:
This is not new code. I simply moved it from another file which has been dissolved.